### PR TITLE
Remove references to CUDA 9.2

### PR DESCRIPTION
**Pull request by janeyx99**
This PR removes references of deprecated 9.2 from the codebase. Most of it was a command F and delete, but some notable changes are:

- Removing references to 9.2 + older specific code in CMake. I started removing some other older version logic, but stopped for FindCUDA.cmake and Dependencies.cmake as they should be able to handle when the user has older versions of CUDA.
- Removing C10_HOST_CONSTEXPR and replacing it with constexpr

### CODE REVIEW

File: .circleci/scripts/binary_windows_build.sh [hunk 1]
```diff
@@ -8,12 +8,7 @@ export CUDA_VERSION="${DESIRED_CUDA/cu/}"
 export USE_SCCACHE=1
 export SCCACHE_BUCKET=ossci-compiler-cache-windows
 export NIGHTLIES_PYTORCH_ROOT="$PYTORCH_ROOT"
-
-if [[ "$CUDA_VERSION" == "92" || "$CUDA_VERSION" == "100" ]]; then
```
Comment (janeyx99) [inline .circleci/scripts/binary_windows_build.sh:12]: Are we still supporting 10.0?

File: .jenkins/pytorch/win-test-helpers/installation-helpers/install_magma.bat [hunk 2]
```diff
@@ -5,15 +5,15 @@ if not "%USE_CUDA%"=="1" (
...
+    echo CUDA version %CUDA_VERSION% format isn't correct, which doesn't contain '.'
     exit /b 1
 )
 
 set VERSION_SUFFIX=%CUDA_VERSION:.=%
 set CUDA_SUFFIX=cuda%VERSION_SUFFIX%
 
 if "%CUDA_SUFFIX%" == "" (
-  echo unknown CUDA version, please set `CUDA_VERSION` higher than 9.2
+  echo unknown CUDA version, please set `CUDA_VERSION` higher than 10.2
```
Comment (janeyx99) [inline .jenkins/pytorch/win-test-helpers/installation-helpers/install_magma.bat:16]: This needs to be replaced with whatever the minimum version actually is.

File: README.md [hunk 3]
```diff
@@ -174,7 +174,7 @@ You will get a high-quality BLAS library (MKL) and you get controlled dependency
 Once you have [Anaconda](https://www.anaconda.com/distribution/#download-section) installed, here are the instructions.
 
 If you want to compile with CUDA support, install
-- [NVIDIA CUDA](https://developer.nvidia.com/cuda-downloads) 9.2 or above
+- [NVIDIA CUDA](https://developer.nvidia.com/cuda-downloads) 10.1 or above
```
Comment (janeyx99) [inline README.md:177]: Same with the minimum version here

File: cmake/Modules_CUDA_fix/upstream/FindCUDA.cmake [hunk 4]
```diff
@@ -981,62 +981,35 @@ endmacro()
 
 #######################
 # Disable emulation for v3.1 onward
-if(CUDA_VERSION VERSION_GREATER "3.0")
-  if(CUDA_BUILD_EMULATION)
-    message(FATAL_ERROR "CUDA_BUILD_EMULATION is not supported in version 3.1 and onwards.  You must disable it to proceed.  You have version ${CUDA_VERSION}.")
-  endif()
+if(CUDA_BUILD_EMULATION)
```
Comment (janeyx99) [inline cmake/Modules_CUDA_fix/upstream/FindCUDA.cmake:984]: I think we should just get rid of CUDA_BUILD_EMULATION entirely.

File: test/cpp/jit/test_gpu.cpp [hunk 5]
```diff
@@ -2609,7 +2609,7 @@ TEST(NVFuserTest, FusionUnaryOps_CUDA) {
   // [Note: explicit tuple type for uniform initialization list]
   // Tuple type must be explicitly specified for each uniform initialization
   // list within the vector to make this code compatible with some old env
-  // which we still need to support. eg. gcc 5.4 + cuda 9.2.
+  // which we still need to support. e.g., gcc 5.4 + cuda 9.2.
```
Comment (janeyx99) [inline test/cpp/jit/test_gpu.cpp:2612]: Do we still want the explicit types here?

File: torch/autograd/profiler.py [hunk 6]
```diff
@@ -654,8 +654,8 @@ def parse_nvprof_trace(path):
     unique = EnforceUnique()
     for row in conn.execute(kernel_query):
         unique.see(row['marker_id'], row['runtime_id'])
-        # 211 is cudaKernelLaunch for cuda >= 9.2; 13 is for older cuda versions
-        assert (row['cbid'] == 211) or (row['cbid'] == 13)
```
Comment (janeyx99) [inline torch/autograd/profiler.py:658]: What is cbid and should this be a different number?

**Comment by janeyx99** [comment]
The 10.1 failure is expected--should be fine after I rebase onto the landed version of https://github.com/pytorch/pytorch/pull/64658#issuecomment-915471660

**Review by malfet — APPROVED** [review]

**Review by seemethere — APPROVED** [review]

**Comment by janeyx99** [comment]
This PR has been broken apart into more manageable pieces:
#65024 #65059 #65065 #65066 #65070

**Comment by janeyx99** [comment]
Closing this as the progress will be continued in other PRs.

**Pull request closed without merging**